src/functions/module_math.ts

```diff
@@ -1317,3 +1318,68 @@ export const TRUNC: AddFunctionDescription = {
   },
   isExported: true,
 };
+
+export const MUNIT: AddFunctionDescription = {
+  description: _lt("Return an n*n identity matrix."),
+  args: [arg("n (number)", _lt("size of the unit matrix"))],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (n: PrimitiveArgValue): number[][] {
+    return Array.from({ length: toNumber(n) }, (_, i) =>
+      Array.from({ length: toNumber(n) }, (_, j) => (i == j ? 1 : 0))
+    );
+  },
+  isExported: true,
+};
+
+export const MFILL: AddFunctionDescription = {
+  description: _lt("Return an n*n matrix filled with n."),
+  args: [
+    arg("n (number)", _lt("number of column of the matrix")),
+    arg("m (number)", _lt("number of row of the matrix")),
+    arg("v (number)", _lt("value to fill matrix")),
+  ],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (n: PrimitiveArgValue, m: PrimitiveArgValue, v: PrimitiveArgValue): any[][] {
+    return Array.from({ length: toNumber(n) }, (_, i) =>
+      Array.from({ length: toNumber(m) }, (_, j) => v)
+    );
+  },
+  isExported: true,
+};
+
+export const RANDARRAY: AddFunctionDescription = {
+  description: _lt("Return an n*n identity matrix."),
+  args: [
+    arg("n (number, default=1)", _lt("size of the unit matrix")),
+    arg("m (number, default=1)", _lt("size of the unit matrix")),
+  ],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (n: PrimitiveArgValue = 1, m: PrimitiveArgValue = 1): number[][] {
+    return Array.from({ length: toNumber(n) }, (_, i) =>
+      Array.from({ length: toNumber(m) }, (_, j) => Math.random())
+    );
+  },
+  isExported: true,
+};
+
+function transpose(values, callback: (value: any) => any) {
+  if (isMatrix(values)) {
+    return Array.from({ length: values[0].length }, (_, i) =>
+      Array.from({ length: values.length }, (_, j) => callback(values[j][i]))
+    );
+  }
+  return callback(values);
+}
+
+export const TRANSPOSE: AddFunctionDescription = {
+  description: _lt("Transpose a matrix."),
+  args: [arg("matrix (range<number>)", _lt("The matrix to be transposed."))],
+  returns: ["NUMBER"],
```

Comment:
return type. 

But to be faire, is the return type even used anywhere ? We have function like `POW` with wrong return type since day one....